### PR TITLE
Add/examples

### PR DESCRIPTION
**Pull request by tj**

### CODE REVIEW

**Comment by jonathanong** [comment]
@brentburgoyne is order preserved when yielding the channel? should go in the read me since i'm not familiar with go yet. i think both order preserving and not are both valid use cases, though i guess not order preserving would be the better default.

**Comment by tj** [comment]
AFAIK they're FIFO, maybe some exceptions?

**Comment by jonathanong** [comment]
yeah that's what i'm guessing. man, this library would make the multipart parser so much easier.

**Comment by tj** [comment]
i really like this approach for utilizing existing callbacks though that's a nice win, though I'd still rather do `yield read(file)` than `fs.readFile(file, ch); yield ch` but there's a lot of use-cases where this will be nice

File: examples/requests.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  var ch = chan();
+
+  urls.forEach(function(url){
+    request.get(url, ch);
+  });
+
+  var res;
+  while (res = yield ch) {
+    console.log(res.status);
+  }
```
Comment (jonathanong) [inline examples/requests.js:23]: wouldn't it just hang here? it'll wait until the next time `ch()` is called. it needs a (optional) counter IMO so it knows when the channel is empty.

``` js
urls.forEach(function(url){
  request.get(url, ch.push());
});

var res;
while (res = yield ch.shift()) {
  console.log(res.status);
}
console.log('i am done');
```

this isn't necessary if you're executing stuff in a `co()` instance where you just want to make sure errors don't occur and can just let it hang.

Comment (brentropy) [inline examples/requests.js:23]: Go has the concept of closing channels. I might do something like this when I get the time:

``` javascript
// ...
urls.forEach(function(url) {
  request.get(url, ch);
});

ch.close();

while (ch.sending()) {
  console.log(yield ch);
}
console.log('i am done');
// ...
```

Once the channel has been closed and the array has been emptied, the `sending()` method would return false.

**Comment by brentropy** [comment]
@jonathanong: No, the order is not preserved. I thought about having an option to preserve order, but I think that if that is a requirement a channel is not the right tool for the job.

File: examples/requests.js (hunk 1 above)
Comment (brentropy) [inline examples/requests.js:23]: I have a [pull request](https://github.com/brentburgoyne/chan/pull/6) with the `close()` method. I am hoping to get some feedback on the implementation if have time to look at it.

**Pull request merged**